common/containers/Tabs/SendTransaction/components/AmountField.jsx

```diff
@@ -64,8 +78,33 @@ export default class AmountField extends React.Component {
   };
 
   onSendEverything = () => {
-    if (this.props.onChange) {
-      this.props.onChange("everything", this.props.unit);
+    const { unit, balance, tokenBalances, onChange } = this.props;
+    if (unit === 'ether') {
+      // TODO sub gas for eth
+      return onChange && onChange(balance.toString(), unit);
+    }
+
+    const token = tokenBalances.find(token => token.symbol === unit);
+    if (!token) {
+      return;
     }
+    return onChange && onChange(token.balance.toString(), unit);
   };
 }
+
+function mapStateToProps(state: AppState) {
+  const tokenBalances = getTokenBalances(state);
+  const tokens = tokenBalances
```

Comment:
can we make it a selector and isolate the logic? Will also help for unit tests eventually.